### DISCUSSION TITLE
Add a metric for the difficulty of a map

### DISCUSSION THREAD

**Discussion by Sogolumbo** [beat-feet/beat-feet#83] [Ideas]
E.g. on the maintenance map the first part are three consecutive lanterns that are 3 blocks high. This is really hard to get over without losing life (you'd have to nail the jump onto the first lantern and then also get over the next two lanterns, if that's even possible).

It would be cool if we were able to find difficult situations, find rules for difficulty and then use this difficulty:
- to reorder the levels
- to show next to the levels
- to find difficult parts in the levels
- to improve the map generation by e.g. setting a 'goal difficulty' and changing map parameters (e.g. sensitivity to the music, amount of obstacles per distance) to achieve that difficulty. 
(Additionally it might be nice to add a slight increase of difficulty over the course of each song or introduce simpler sections with checkpoints)

**Comment by pserwylo** [comment]
Hmm, so this is all done manually now. In the (closed) issue #50 you can read a bit of my internal thoughts on why the levels are the way they are in terms of order.

May I ask though, are you able to utilise the double jump to get over these three posts? It was a little tricky, but I captured two screenshots of how I would do this:

**Firstly,** double jump onto the first pole.

![image](https://user-images.githubusercontent.com/248565/112793341-dd842180-90b0-11eb-919e-a9bcec777120.png)

**Secondly,** once you land on the pole, double jump again to either land onto the second pole, or jump right over it.

![image](https://user-images.githubusercontent.com/248565/112793350-e1b03f00-90b0-11eb-8139-bb8e0ad4a06a.png)

Hopefully that helps!

**Reply by Sogolumbo** [reply]
I think I could pull this off with a little practise, but I'm currently not in the mood to do so.
I was able to get onto the first lantern sometimes. After that I usually fell in between the second and third lantern.
